HIDS samples: add continuous report sending/receiving feature (#28697)
nrfconnect/sdk-nrf: samples/bluetooth/peripheral_hids_keyboard/README.rst

      LED 4:
         Indicates if an NFC field is present.

      Buttons 1 and 3 together:
         Toggle **continuous report sending** when :option:`CONFIG_SAMPLE_BT_HIDS_CONTINUOUS_REPORT_SENDING` is enabled (letter / passkey-confirm button and third physical button pressed at the same time).
         See :ref:`peripheral_hids_keyboard_continuous_tx` for more details.
         .. note::

>             Pressing buttons 1 and 2 will be ignored until the continuous report sending is turned off.

Reviewer: ```suggestion
         Toggle continuous report sending when the :option:`CONFIG_SAMPLE_BT_HIDS_CONTINUOUS_REPORT_SENDING` Kconfig option is enabled (letter/passkey-confirm button and third physical button pressed at the same time).
         See :ref:`peripheral_hids_keyboard_continuous_tx` for more details.

         .. note::

            Pressing buttons **1** and **2** is ignored until the continuous report sending is turned off.
```

      LED 3:
         Indicates if an NFC field is present.

      Buttons 0 and 2 together:
         Toggle **continuous report sending** when :option:`CONFIG_SAMPLE_BT_HIDS_CONTINUOUS_REPORT_SENDING` is enabled (letter / passkey-confirm button and third physical button pressed at the same time).
         See :ref:`peripheral_hids_keyboard_continuous_tx` for more details.
         .. note::

>             Pressing buttons 0 and 1 will be ignored until the continuous report sending is turned off.

Reviewer: ```suggestion
         Toggle continuous report sending when the :option:`CONFIG_SAMPLE_BT_HIDS_CONTINUOUS_REPORT_SENDING` Kconfig option is enabled (letter/passkey-confirm button and third physical button pressed at the same time).
         See :ref:`peripheral_hids_keyboard_continuous_tx` for more details.

         .. note::

            Pressing buttons **0** and **1** is ignored until the continuous report sending is turned off.
```

.. options-from-kconfig::
   :show-type:

.. _peripheral_hids_keyboard_continuous_tx:

Continuous report sending
-------------------------

> The :option:`CONFIG_SAMPLE_BT_HIDS_CONTINUOUS_REPORT_SENDING` Kconfig option enables sending HID keyboard input notifications at roughly **one per connection interval**, timed with the SoftDevice Controller **radio notification** API (:kconfig:option:`CONFIG_BT_RADIO_NOTIFICATION_CONN_CB`).

Reviewer: ```suggestion
The :option:`CONFIG_SAMPLE_BT_HIDS_CONTINUOUS_REPORT_SENDING` Kconfig option enables sending HID keyboard input notifications at roughly one for each connection interval, timed with the SoftDevice Controller radio notification API (:kconfig:option:`CONFIG_BT_RADIO_NOTIFICATION_CONN_CB`).
```

> Follow the :ref:`peripheral_hids_keyboard_user_interface` section to see how to toggle the feature on or off.

Reviewer: ```suggestion
See the :ref:`peripheral_hids_keyboard_user_interface` section for how to toggle the feature on or off.
```

> If continuous report sending is enabled, any button press other than the designated button combination will not send key presses to the host until the continuous report sending is turned off.

Reviewer: ```suggestion
If continuous report sending is enabled, any button press other than the designated button combination does not send key presses to the host until the continuous report sending is turned off.
```

> Whenever the continuous report sending is halted, statistics are displayed: the total number of connection intervals, the number of reports that were not sent on time, and the number of connection intervals with no report sent.

Reviewer: ```suggestion
Whenever the continuous report sending is halted, the following statistics are displayed:

* The total number of connection intervals.
* The number of reports that were not sent on time.
* The number of connection intervals with no report sent.

```

To print interval statistics on the central side, run the :ref:`bluetooth_central_hids` sample against this peripheral.

> The feature is not supported in Boot Protocol Mode.

Reviewer: ```suggestion
The feature is not supported in the Boot Protocol mode.
```
Does this mode need to be in initial caps?

> If more than one central is connected, only one central is targeted for this traffic; which link is used is not specified - use a single connection for predictable measurements.

Reviewer: ```suggestion
If more than one central is connected, only one central is targeted for this traffic.
Which link is used is not specified - use a single connection for predictable measurements.
```

> For best results, only allow a single central connection (set both :kconfig:option:`CONFIG_BT_MAX_CONN` and :kconfig:option:`CONFIG_BT_HIDS_MAX_CLIENT_COUNT` to 1).

Reviewer: ```suggestion
For best results, only allow a single central connection (set both :kconfig:option:`CONFIG_BT_MAX_CONN` and :kconfig:option:`CONFIG_BT_HIDS_MAX_CLIENT_COUNT` to ``1``).
```

> Otherwise more reports will be dropped or delayed due to other radio traffic such as advertising events.

Reviewer: ```suggestion
Otherwise, more reports will be dropped or delayed due to other radio traffic, such as advertising events.
```
